bindings/java/src/org/sleuthkit/datamodel/OsAccount.java

```diff
@@ -0,0 +1,439 @@
+
+		OsAccountInstanceType(int id, String name) {
+			this.id = id;
+			this.name = name;
+		}
+
+		/**
+		 * Get account instance type id.
+		 *
+		 * @return Account instance type id.
+		 */
+		public int getId() {
+			return id;
+		}
+
+		/**
+		 * Get account instance type name.
+		 *
+		 * @return Account instance type name.
+		 */
+		public String getName() {
+			return name;
+		}
+
+		/**
+		 * Gets account instance type enum from id.
+		 *
+		 * @param typeId Id to look for.
+		 *
+		 * @return Account instance type enum.
+		 */
+		public static OsAccountInstanceType fromID(int typeId) {
+			for (OsAccountInstanceType statusType : OsAccountInstanceType.values()) {
+				if (statusType.ordinal() == typeId) {
+					return statusType;
+				}
+			}
+			return null;
+		}
+	}
+
+	/**
+	 * Constructs an OsAccount with a realm/username and unique id, and
+	 * signature.
+	 *
+	 * @param sleuthkitCase  The SleuthKit case (case database) that contains
+	 *                       the artifact data.
+	 * @param osAccountobjId Obj id of the account in tsk_objects table.
+	 * @param realm	         Realm - defines the scope of this account.
+	 * @param loginName      Login name for the account. May be null.
+	 * @param uniqueId       An id unique within the realm - a SID or uid. May
+	 *                       be null, only if login name is not null.
+	 * @param signature	     A unique signature constructed from realm id and
+	 *                       loginName or uniqueId.
+	 */
+	OsAccount(SleuthkitCase sleuthkitCase, long osAccountobjId, OsAccountRealm realm, String loginName, String uniqueId, String signature) {
+		
+		super(sleuthkitCase, osAccountobjId, signature);
+		
+		this.sleuthkitCase = sleuthkitCase;
+		this.osAccountobjId = osAccountobjId;
+		this.realm = realm;
+		this.loginName = loginName;
+		this.uniqueId = uniqueId;
+		this.signature = signature;
+	}
+
+	/**
+	 * Sets the account user's full name.
+	 *
+	 * @param fullName Full name.
+	 */
+	public void setFullName(String fullName) {
+		this.fullName = fullName;
+	}
+
+	/**
+	 * Sets the admin flag for the account.
+	 *
+	 * @param isAdmin Flag to indicate if the account is an admin account.
+	 */
+	public void setIsAdmin(boolean isAdmin) {
+		this.isAdmin = isAdmin;
+	}
+
+	/**
+	 * Sets account type for the account.
+	 *
+	 * @param osAccountType Account type..
+	 */
+	public void setOsAccountType(OsAccountType osAccountType) {
+		this.osAccountType = osAccountType;
+	}
+
+	/**
+	 * Sets account status for the account.
+	 *
+	 * @param osAccountStatus Account status.
+	 */
+	public void setOsAccountStatus(OsAccountStatus osAccountStatus) {
+		this.osAccountStatus = osAccountStatus;
+	}
+
+	/**
+	 * Set account creation time.
+	 *
+	 * @param creationTime Creation time.
+	 */
+	public void setCreationTime(long creationTime) {
+		this.creationTime = creationTime;
+	}
+
+	/**
+	 * Set the signature.
+	 * 
+	 * signature may changes if the login name or unique id is updated after creation.
```

Comment:
signature may changes => The signature may change

Reply:
done